test/integration/daemonset/daemonset_test.go

```diff
@@ -524,7 +523,7 @@ func TestOneNodeDaemonLaunchesPod(t *testing.T) {
 			defer close(stopCh)
 
 			// Start Scheduler
-			setupScheduler(t, clientset, informers, stopCh)
+			defer setupScheduler(t, clientset, informers, stopCh)()
```

Comment:
it seems that feature gate value gets reset to its original value at then, if that is the case wouldn't just executing ApplyFeatureGates again be sufficient? something like defer ApplyFeatureGates()

Reply:
We can do it that way, but that needs to accomplish `else` block for each `if` block in this function. Which could be error-prone. And in caller side, when you call it twice, you can't simply `defer ApplyFeatureGates()`, you have to change the call chain like this:

```go
f := featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.TaintNodesByCondition, true)
ApplyFeatureGates()

defer func() {
	f()
	ApplyFeatureGates()
}
```

IMO this's not easy reading. And this style of code will violate a CI check: hack/verify-test-featuregates.sh. You have to fix that as well...

Reply:
oh, defer is stacked so we can't just defer the ApplyFeatureGates. OK, keeping the current approach sounds good then.